Terraform secret engine docs - support multiple team tokens (#30477)
hashicorp/vault: website/content/docs/secrets/terraform.mdx

token_id        at-fqvtdTQ5kQWcjUfG
```

### Team roles

The HCP Terraform secret engine creates dynamic Team API tokens by 
configuring a Vault role to manage an existing HCP Terraform Team. 
The lifecycle of these tokens is managed by Vault and/or HCP Terraform.

> Generally, Vault aims to manage tokens in external APIs with revoke actions
> taken according to ttl & max_ttl settings. In addition to that behavior,
> using the max_ttl Vault will set an `ExpiredAt` date in HCP Terraform 
> also which will ensure the token expires at the Max TTL time. This prevents 
> Team tokens living past their max ttl if Vault is unable to revoke the token.
> Omitting a max_ttl will Vault's system max_ttl

Reviewer: The last sentence appeared to be incomplete?

```suggestion
Generally, Vault aims to manage tokens in external APIs with revoke actions
taken according to `ttl` and `max_ttl` settings. In addition to that behavior,
using the `max_ttl` Vault will set an `ExpiredAt` date in HCP Terraform 
which will ensure the token expires at the Max TTL time. This prevents 
Team tokens living past their `max_ttl` if Vault is unable to revoke the token.
Omitting the `max_ttl` value will default to Vault's system `max_ttl`.
```

Author: roasted 😭 

(jk) appreciate your thorough review :) keep um coming!

In HCP Terraform, team tokens cannot have matching descriptions. In order to avoid

> collissions, we generate a random string as a suffix to the description. It is

Reviewer: ”We" usually means HashiCorp. In this case, "Vault" or perhaps, "this secrets engine" maybe?

```suggestion
collissions, Vault generates a random string as a suffix to the description. It is
```

Author: ```suggestion 
collisions, the secret engine generates a random string as a suffix to the description. It is
```
